de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/general/UiSnippetRoleManagement.java

```diff
@@ -76,19 +89,53 @@ protected String getSectionDescription() {
 		return SECTION_DESCRIPTION_PREFIX + user;
 	}
 
+    /**
+     * Open a custom dialog to manage users for a discipline.
+     */
+	private void openCustomDialog() {
+	    Table table = tableViewer.getTable();
+	    int selectionIndex = table.getSelectionIndex();
+
+	    if (selectionIndex != -1) {
+	        Discipline selectedDiscipline = (Discipline) tableViewer.getElementAt(selectionIndex);
+
+	        // Extract the users for the selected discipline
+	        String[] existingUsernames = selectedDiscipline.getUsers().toArray(new String[0]);

```

Comment:
Why are you actually using an array here? You could just leave it in the list right? Then you actually have more functionallity as with the array - and you don't need to copy the data which slows the application down

Reply:
I would agree on that with Tobi